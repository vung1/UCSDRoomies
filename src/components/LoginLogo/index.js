import React from "react";
import { Text, Image, View, StyleSheet } from "react-native";

const appLogoPNG = require("../../../assets/images/appLogo.png");

function LoginLogo() {
  return (
    <View style={styles.logoContainer}>
      <Image source={appLogoPNG} style={styles.image} />
      <Text style={{ fontSize: 24, fontFamily: "AmericanTypewriter-Bold" }}>
        {" "}
        UCSD{" "}
      </Text>
      <Text style={{ fontSize: 24, fontFamily: "AmericanTypewriter-Bold" }}>
        {" "}
        Roomies{" "}
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  logoContainer: {
    width: 200,
    height: 200,
    backgroundColor: "white",
    alignItems: "center",
    borderRadius: 100,
  },
  image: {
    width: 90,
    height: 90,
    marginTop: "8%",
    marginBottom: "3%",
  },
});

export default LoginLogo;
