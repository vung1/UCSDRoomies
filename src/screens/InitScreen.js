import React from 'react';
import { View, StyleSheet } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

import Logo from '../components/Logo'

const InitScreen = () => {
  return (
    <LinearGradient colors={["#74AED6", "#247DCF"]} style={styles.background}>
      <View style={styles.logoContainer}><Logo/></View>
    </LinearGradient>
  );
}
const styles = StyleSheet.create({
  background: { 
    flex: 1,
    width: '100%',
    alignItems: "center",
  },
  logoContainer: {
    marginTop: "42%",
  }
});

export default InitScreen;
