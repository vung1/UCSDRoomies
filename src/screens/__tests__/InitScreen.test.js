import React from "react";

import {
  render,
  fireEvent,
  screen,
  act,
  waitFor,
  mount,
} from "@testing-library/react-native";

import * as renderer from "react-test-renderer";

import InitScreen from "../InitScreen";
import LoginLogo from "../../components/LoginLogo";

describe("Rendering Tests", () => {
  test("matches snapshot", async () => {
    jest.useFakeTimers();
    const navigation = { navigate: () => {} };
    jest.spyOn(navigation, "navigate");
    // jest.spyOn(navigation, "navigate");
    const tree = renderer
      .create(<InitScreen navigation={navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();

    // Advance the timer by 4 seconds to simulate the timeout
    jest.advanceTimersByTime(1000);

    // Verify that the navigation occurs after the timeout
    await expect(navigation.navigate).toHaveBeenCalledWith(
      "LoginScreen",
      "LoginScreen",
    );
  });
});
