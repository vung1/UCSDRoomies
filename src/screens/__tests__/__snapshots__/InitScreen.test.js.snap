// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Tests matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4285837014,
          4280581583,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": "42%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 100,
            "height": 200,
            "width": 200,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 90,
              "marginBottom": "3%",
              "marginTop": "8%",
              "width": 90,
            }
          }
        />
        <Text
          style={
            Object {
              "fontFamily": "AmericanTypewriter-Bold",
              "fontSize": 24,
            }
          }
        >
           
          UCSD
           
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "AmericanTypewriter-Bold",
              "fontSize": 24,
            }
          }
        >
           
          Roomies
           
        </Text>
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
